pilot/pkg/security/model/authentication.go

```diff
@@ -125,7 +126,7 @@ func ConstructSdsSecretConfig(name, sdsUdsPath string) *tls.SdsSecretConfig {
 					},
 				},
 			},
-			InitialFetchTimeout: features.InitialFetchTimeout,
+			InitialFetchTimeout: ptypes.DurationProto(time.Second * 0),
```

Comment:
nit: warrants a comment why we do *not* follow `features.InitialFetchTimeout` but still do in the other usages. Specifically, that we want it for gateway certs which may or may not exist, but not for workload certs which must exist

Reply:
@adiprerepa please add a brief explanation above this line as @howardjohn commented.

Reply:
I am not sure if I full understand it
Is that listener at ingressgateway should set features.InitialFetchTimeout while cluster at ingressgateway(and for all sidecars) should set infinite initialfetchtimeout?

Reply:
+1 - infinite doesn't sound good.. Comment explaining the problem, solution would help people who'll work on this in the future.

Reply:
If we are using TLS I would rather a pod fail to startup then to startup without certs and suddenly lead to random traffic failures. 100% agree on comments though

Reply:
sure ill add the comment

Reply:
this is going to take a while, so many tests fail with this change.

Reply:
Is there a ETA for this? It would be great if this could be shipped with OSS 1.7. https://github.com/istio/istio/wiki/Istio-Release-1.7#timeline

Reply:
ill try to get this shipped @JimmyCYJ by 1.7